apollo-federation/src/schema/field_set.rs

```diff
@@ -57,15 +57,25 @@ pub(crate) fn parse_field_set(
     schema: &ValidFederationSchema,
     parent_type_name: NamedType,
     field_set: &str,
+    validate: bool,
 ) -> Result<SelectionSet, FederationError> {
     // Note this parsing takes care of adding curly braces ("{" and "}") if they aren't in the
     // string.
-    let field_set = FieldSet::parse_and_validate(
-        schema.schema(),
-        parent_type_name,
-        field_set,
-        "field_set.graphql",
-    )?;
+    let field_set = if validate {
+        FieldSet::parse_and_validate(
+            schema.schema(),
+            parent_type_name,
+            field_set,
+            "field_set.graphql",
```

Comment:
Would it not be better to leave the filename blank?

Reply:
I mainly did it because other `parse_and_validate()` calls were doing it, but the other reason is that the empty filename [isn't technically a legal UNIX filename](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_170), and if `apollo-compiler` at some point uses a library that expects filenames to be valid, you may get an unexpected error.